### PR TITLE
Alfoa/validation gate

### LINKED ISSUE #1513: [TASK] Addition of a standardized API to add and deploy validation algorithms
--------
Issue Description
--------

**Is your feature request related to a problem? Please describe.**
The development of validation algorithms requires the deployement of a validation pp that can function as "gate" for the different algorithms

**Describe the solution you'd like**
development of a new PP that targets Validation processes

**Describe alternatives you've considered**
None

validation_pp_design
----------------
For Change Control Board: Issue Review
----------------
This review should occur before any development is performed as a response to this issue.
- [x] 1. Is it tagged with a type: defect or task?
- [x] 2. Is it tagged with a priority: critical, normal or minor?
- [x] 3. If it will impact requirements or requirements tests, is it tagged with requirements?
- [x] 4. If it is a defect, can it cause wrong results for users? If so an email needs to be sent to the users.
- [x] 5. Is a rationale provided? (Such as explaining why the improvement is needed or why current code is wrong.)

-------
For Change Control Board: Issue Closure
-------
This review should occur when the issue is imminently going to be closed.
- [x] 1. If the issue is a defect, is the defect fixed?
- [x] 2. If the issue is a defect, is the defect tested for in the regression test system?  (If not explain why not.)
- [x] 3. If the issue can impact users, has an email to the users group been written (the email should specify if the defect impacts stable or master)?
- [x] 4. If the issue is a defect, does it impact the latest release branch? If yes, is there any issue tagged with release (create if needed)?
- [x] 5. If the issue is being closed without a pull request, has an explanation of why it is being closed been provided?

### LINKED ISSUE #1514: [DEFECT] Outstream error if not Dataobject found results in a crash
--------
Defect Description
--------

**Describe the defect**

##### What did you expect to see happen?
The OutStreamInterface to error out with a meaningful error if a DataObject that was supposed to be in the step, was not inputted by the user.

##### What did you see instead?
A crash:
image001

##### Do you have a suggested fix for the development team?
Fix of the index that is not defined

**Describe how to Reproduce**
1. 
2. 
3. 
4. 

**Screenshots and Input Files**
Please attach the input file(s) that generate this error.  The simpler the input, the faster we can find the issue.

**Platform (please complete the following information):**
 - Version: [e.g. 22]
 - Dependencies Installation: [CONDA or PIP]

----------------
For Change Control Board: Issue Review
----------------
This review should occur before any development is performed as a response to this issue.
- [x] 1. Is it tagged with a type: defect or task?
- [x] 2. Is it tagged with a priority: critical, normal or minor?
- [x] 3. If it will impact requirements or requirements tests, is it tagged with requirements?
- [x] 4. If it is a defect, can it cause wrong results for users? If so an email needs to be sent to the users.
- [x] 5. Is a rationale provided? (Such as explaining why the improvement is needed or why current code is wrong.)

-------
For Change Control Board: Issue Closure
-------
This review should occur when the issue is imminently going to be closed.
- [x] 1. If the issue is a defect, is the defect fixed?
- [x] 2. If the issue is a defect, is the defect tested for in the regression test system?  (If not explain why not.)
- [x] 3. If the issue can impact users, has an email to the users group been written (the email should specify if the defect impacts stable or master)?
- [x] 4. If the issue is a defect, does it impact the latest release branch? If yes, is there any issue tagged with release (create if needed)?
- [x] 5. If the issue is being closed without a pull request, has an explanation of why it is being closed been provided?

### PR DESCRIPTION
**Pull request by alfoa**
--------
Pull Request Description
--------
##### What issue does this change request address? (Use "#" before the issue to link it, i.e., #42.)
Closes #1513 Closes #1514

##### What are the significant changes in functionality due to this change request?
This PR is in charge of adding a new Post Processor that is aimed to function as a gate for
Validation algorithms. Only one validation algorithm has been added as example (Probabilistic).
It might create some redondancy and it will be removed once the DSS, representativity and PCM are implemented.

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code.
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [x] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.
- [x] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.
- [x] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [x] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [x] 8. If an analytic test is changed/added is the the analytic documentation updated/added?
- [x] 9. If any test used as a basis for documentation examples (currently found in `raven/tests/framework/user_guide` and `raven/docs/workshop`) have been changed, the associated documentation must be reviewed and assured the text matches the example.

### CODE REVIEW

File: framework/Metrics/__init__.py [hunk 1]
```diff
@@ -27,6 +25,7 @@
 from .PairwiseMetric import PairwiseMetric
 from .CDFAreaDifference import CDFAreaDifference
 from .PDFCommonArea import PDFCommonArea
+#from .RACDistance import RACDistance
```
Comment (alfoa) [inline framework/Metrics/__init__.py:28]: I would like to keep this line commented out because, I woul like to add the new metric once this PR is merged

**Review by Jimmy-INL — APPROVED** [review]

**Comment by Jimmy-INL** [comment]
Some regression test machines have connectivity issues currently under investigation by @joshua-cogliati-inl . The tests have been run locally and are green.

The PR is approved and can be merged.

**Comment by Jimmy-INL** [comment]
Some regression test machines have connectivity issues currently under investigation by @joshua-cogliati-inl . The tests have been run locally and are green.

The PR is approved and can be merged.

**Review by Jimmy-INL — DISMISSED** [review]

File: framework/Models/PostProcessors/Validation.py [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+from .PostProcessor import PostProcessor
+from . import validationAlgorithms
+from utils import utils, mathUtils
+from utils import InputData, InputTypes
+import DataObjects
+import MetricDistributor
+#Internal Modules End-----------------------------------------------------------
+
+
+class Validation(PostProcessor):
```
Comment (wangcj05) [inline framework/Models/PostProcessors/Validation.py:38]: please update the base class "PostProcessor" to "PostProcessorInterface"

Comment (alfoa) [reply framework/Models/PostProcessors/Validation.py]: Done

**Comment by alfoa** [comment]
@wangcj05 or @Jimmy-INL  can you approve this and merge it?

**Review by Jimmy-INL — APPROVED** [review]

**Pull request merged**